plugins/src/main/java/org/smartregister/fhir/gateway/plugins/PermissionAccessChecker.java

```diff
@@ -413,6 +414,18 @@ private static Map<String, List<String>> collateSyncStrategyIds(
                                     : new ArrayList<>();
 
                     resultMap = Map.of(syncStrategy, locationIds);
+                } else if (Constants.RELATED_ENTITY_LOCATION.equalsIgnoreCase(syncStrategy)) {
```

Comment:
This should be a separate if statement _(rather than else if)_ so that incase the newly processed `this.syncStrategyIds` is empty the exception is still thrown and logged.